core/base/common/ArrayLinkedList.h

```diff
@@ -0,0 +1,37 @@
+/// \ingroup base
+/// \class ttk::ArrayLinkedList
+/// \author Eve Le Guillou <eve.le-guillou@lip6.fr>
+/// \date Mars 2022.
+
+#include <array>
+#include <list>
+
+#pragma once
+
+namespace ttk {
+  template <typename datatype, int size>
+  class ArrayLinkedList {
+  public:
+    std::list<std::array<datatype, size>> list;
+    int numberOfElement;
+    // In order to prevent false sharing when creating a
+    // std::vector of ArrayLinkedList objects (one element
+    // of the std::vector for each thread), it is necessary
+    // that one ArrayLinkedList object be bigger than one cache
+    // line. Here, we assume a cache line to be 64 bytes.
+    unsigned char padding[32];
```

Comment:
I guess it'd be preferrable to replace that with a modern variant:
`std::array<unsigned char, 32> padding_{};`